src/main/java/com/avispl/symphony/dal/communicator/management/zoom/rooms/ZoomRoomsCommunicator.java

```diff
@@ -0,0 +1,581 @@
+     */
+    private Map<String, String> parseZoomRoomsProperties(String response, String propertyIdentifier) {
+        Map<String, String> parameters = new HashMap<>();
+        Arrays.asList(response.split("\n")).forEach(s -> {
+            if (s.startsWith(propertyIdentifier)) {
+                String[] values = s.split(":");
+                parameters.put(values[0].replace("*s ", ""), values[1].replace("\r", "").trim());
+            }
+        });
+        return parameters;
+    }
+
+    /**
+     * Join meeting that is currently in progress or start a scheduled one.
+     *
+     * @param value - contains dialString of a format %meetingNumber%.%meetingPassword%@zoomcrc.com
+     *              password is optional.
+     * @return meetingId if operation is successful
+     */
+    private String joinStartMeeting(String value) throws Exception {
+        boolean hasPassword;
+        String meetingNumber;
+        String meetingPassword;
+
+        Pattern dialStringPattern = Pattern.compile("(\\d+)\\.?(\\d+)?(@[a-z]+?\\.[a-z]{2,5})");
+        Matcher matcher = dialStringPattern.matcher(value);
+        if (!matcher.find() && !matcher.matches()) {
+            // This might be the section to call out SIP/h323 devices it "invite" functionality fits our needs
+            // However, for the simplified approach, this option is skipped for now
+            throw new OperationNotSupportedException("Dial string does not match expected pattern meetingNumber.meetingPassword@zoomcrc.com");
+        } else {
+            hasPassword = !StringUtils.isNullOrEmpty(matcher.group(2));
+            meetingNumber = matcher.group(1);
+            meetingPassword = hasPassword ? matcher.group(2) : "";
+        }
+
+        if (logger.isDebugEnabled()) {
+            logger.debug(String.format("Attempt to join or start a meeting with meetingNumber: %s and password: %s", meetingNumber, meetingPassword));
+        }
+
+        if (executeAndVerify(String.format(ZCOMMAND_DIAL_JOIN, meetingNumber) + (hasPassword ? (" password:" + meetingPassword) : "") + "\r", "*r DialJoinResult (status=OK)")) {
+            isHost = false;
+            if (logger.isDebugEnabled()) {
+                logger.debug("Joining the meeting: " + meetingNumber);
+            }
+            this.meetingNumber = meetingNumber;
+            return meetingNumber;
+        } else {
+            if (executeAndVerify(String.format(ZCOMMAND_DIAL_START, meetingNumber) + (hasPassword ? (" password:" + meetingPassword) : "") + "\r", "*r DialJoinResult (status=OK)")) {
+                isHost = true;
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Starting the meeting: " + meetingNumber);
+                }
+                this.meetingNumber = meetingNumber;
+                return meetingNumber;
+            }
+        }
+        throw new IllegalStateException("Failed to connect to a meeting with dial string: " + value);

```

Comment:
Not sure it is a correct exception. What state it refers to? State of communicator? Or state of the meeting? And what's illegal in that state?
Also should probably be moved to inside of else above.

Reply:
I figured this would state "the state of the connection", but it may also be IllegalArgumentException, meaning that the dial string does not contain data that can be used for establishing a connection. Given that the SIP/H323 cases were omitted here - latter would make more sense

Reply:
Nope, sorry.
IllegalState is when object asked to do something but it is in such state that whatever was asked cannot be performed. E.g. it asked to perform some business operation without being initialized first. Or after it is being destroyed.
IllegalArgument is when it is asked to do something but cannot do it because one of the passed parameters is invalid.
In this case it is neither. The code actually did something, but verify result returned false. The error must represent what's actually failed. If there is no appropriate exception to represent what is actually failed, and there is no valid reason to create dedicated exception class, then the last resort is to use RuntimeException (something failed at runtime).